### ISSUE TITLE
"Failed to connect socket: Unknown error" from Windows mpd.exe built from git source

### ISSUE THREAD

**Issue by gdub010101** [MusicPlayerDaemon/MPD#134]
I wanted to build mpd for Windows, just to be sure I could do it before attempting to add some 3rd-party libraries to the build (like libmpdclient so I can connect to an existing MPD database).

So on Arch x86_64, I did a 'git clone' of MPD, and from the root of the project ran ./autogen.sh to build configure, and then ran ./win32/build.py --64.  The build ran without errors and resulted in ./src/mpd.exe.

However, when I run this mpd.exe on the Windows 10 machine, I get

```
PS C:\mpdtest> .\mpd.exe .\mpd.conf -v --stderr
config_file: loading file ./mpd.conf
exception: Failed to connect socket: Unknown error
```

If I download the pre-built mpd.exe from https://www.musicpd.org/download/win32/0.20.11/, I don't receive this error.

```
PS C:\mpdtest> .\mpd.exe .\mpd.conf -v --stderr
config_file: loading file ./mpd.conf
vorbis: Xiph.Org libVorbis 1.3.5
opus: libopus 1.2.1
exception: Failed to open C:/mpdtest/mpd.db: The system cannot find the file specified.
: No such file or directory
curl: version 7.55.1
exception: Input plugin 'ffmpeg' is unavailable: No protocol
update: spawned thread for update job id 1
update: starting
simple_db: removing empty directories from DB
simple_db: sorting DB
simple_db: writing DB
update: finished
```

In both cases, my mpd.conf is the same
```

music_directory		"C:/mpdtest"
db_file			"C:/mpdtest/mpd.db"
log_file			"C:/mpdtest/mpd.log"
port				"6601"
log_level			"verbose"

input {
        plugin "curl"
}

audio_output {
	type		"winmm"
	name		"My WinMM output"
}
```

There must be some difference between using win32/build.py from git and how the "official" builds are being done.  Any ideas?

**Comment by MaxKellermann** [comment]
This is exactly how I do the "official" build. Could be a broken toolchain on your side, but I have no idea. Upload your (unstripped) executable, and I'll try to have a look.

**Comment by gdub010101** [comment]
Thanks, [here's the exe, zipped](https://www.dropbox.com/s/b3kya2zrjkefhbj/mpd.zip?dl=0).

Also attached the output of win32/build.py.
[build.log](https://github.com/MusicPlayerDaemon/MPD/files/1409295/build.log)

**Comment by MaxKellermann** [comment]
I found out: you're comparing the released v0.20.x with git master, which is (or will become) v0.21. The code is different, and buggy.
